envsec/pkg/awsfed/awsfed.go

```diff
@@ -129,3 +172,23 @@ func GenSSMConfigForUser(
 		Region:          fed.Region,
 	}, nil
 }
+
+func GenSSMConfigFromIdToken(
```

Comment:
Maybe merge both functions and call them `GenSSMConfigFromToken` 

and add a `useCache` param.